### PR TITLE
dev/customer: Implement Customer Business Logic

### PR DESCRIPTION
**Pull request by sgoudham**

### CODE REVIEW

File: src/main/java/org/somerville/swag/data/entity/state/Guest.java [hunk 1]
```diff
@@ -52,18 +87,47 @@ public void viewBasket() {
...
 
     @Override
-    public void removeProductFromBasket() {
+    public void removeProductFromBasket(OrderLine orderLine) {
+        customer.getCurrentOrder().getOrderLines().remove(orderLine);
+        Product selectedProduct = orderLine.getProduct();
+        int quantity = orderLine.getQuantity();
+        selectedProduct.setStockLevel(selectedProduct.getStockLevel() - quantity);
 
+        // TODO: Write New Product Stock Level to Database
```
Comment (RobertRanson) [inline src/main/java/org/somerville/swag/data/entity/state/Guest.java:105]: Was our design decision to only update product stock level after checkout? so removing item from basket shouldn't require DB?

File: src/main/java/org/somerville/swag/display/Purchase.form [hunk 2]
```diff
@@ -1,99 +1,110 @@
...
-          <grid row="1" column="3" row-span="1" col-span="1" vsize-policy="6" hsize-policy="1" anchor="0" fill="2" indent="0" use-parent-layout="false"/>
-        </constraints>
-      </vspacer>
-      <component id="1032e" class="javax.swing.JLabel">
-        <constraints>
-          <grid row="7" column="0" row-span="3" col-span="1" vsize-policy="0" hsize-policy="0" anchor="8" fill="0" indent="0" use-parent-layout="false"/>
+          <grid row="4" column="0" row-span="1" col-span="3" vsize-policy="0" hsize-policy="3" anchor="0" fill="1" indent="0" use-parent-layout="false"/>
         </constraints>
         <properties>
+          <background color="-12992481"/>
```
Comment (RobertRanson) [inline src/main/java/org/somerville/swag/display/Purchase.form:103]: The design of the JFrame is looking great! nice work!

File: src/test/java/org/somerville/swag/data/entity/CustomerTest.java [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+    }
+
+    @Test
+    void customerStateShouldNotBeNull() {
+        Customer expectedCustomer = new Customer();
+        assertNotNull(expectedCustomer.getCustomerState());
+    }
+
+    @Test
+    void customerDetailsAreRefreshed() {
```
Comment (RobertRanson) [inline src/test/java/org/somerville/swag/data/entity/CustomerTest.java:43]: Excellent testing and very good idea to stub out todo tests

**Review by RobertRanson — APPROVED** [review]

**Review by AlanRobinson1337 — APPROVED** [review]
Mob programming session. Happy with all changes. Go team!

**Review by hzl616 — APPROVED** [review]
A lot of work has gone in here! The refresh method is a great addition to concluding the purchase form. Also excellent work done to adding to the logger again. All of the finer details added to the UI, and information popups make a for a more  informative user experience.

**Review by 2013884f — APPROVED** [review]

**Pull request merged**